deployment/cac-stack.py

```diff
@@ -3,70 +3,50 @@
 import argparse
 import os
-import sys
 
 from cloudformation.stacks import build_stacks
 from cloudformation.template_utils import get_config
 from packer.cac_packer import run_packer
-from auth import get_creds, write_creds, delete_creds
 
 
 file_dir = os.path.dirname(os.path.realpath(__file__))
 
 
-def launch_stacks(cac_config, creds, stack_type, stack_color, **kwargs):
+def launch_stacks(cac_config, stack_type, stack_color, **kwargs):
     """Launches the specified stacks
 
     Args:
       cac_config (Dict): Dictionary of AWS parameter values
-      creds (Dict): Dictionary containing AWS credentials
       stack_type (str): Type of environment (dev, prod, test)
       stack_color (str): Color of environment (blue, green)
     """
 
-    # Launching the stack is a lot easier with the credentials file in place. Otherwise,
-    # there are several places where credentials need to be passed into boto directly.
-    # Write it out temporarily, and remove it afterwards.
-    write_creds(creds, True)
-    try:
-        build_stacks(cac_config, stack_type, stack_color)
-    finally:
-        delete_creds()
+    build_stacks(cac_config, stack_type, stack_color)
 
 
-def create_ami(machine_type, aws_region, creds, aws_config, **kwargs):
+def create_ami(machine_type, aws_region, aws_profile, **kwargs):
     """Creates the specified AMI(s)
 
     Args:
       machine_type (str): Optional type of AMI to build (all are built if unspecified)
       aws_region (str): AWS region id
-      creds (Dict): Dictionary containing AWS credentials
+      aws_profile (str): AWS profile name
     """
-    run_packer(machine_type, aws_region, creds, aws_config)
+    run_packer(machine_type, aws_region, aws_profile)
 
 
 def main():
     """Parse args and run desired commands"""
     common_parser = argparse.ArgumentParser(add_help=False)
     common_parser.add_argument('--aws-region', default='us-east-1',
                                help='AWS region -- defaults to us-east-1')
-    common_parser.add_argument('--aws-access-key-id', required=True,
-                               help='AWS Access Key ID')
-    common_parser.add_argument('--aws-secret-access-key', required=True,
-                               help='AWS Secret Access Key')
-    common_parser.add_argument('--aws-role-arn', required=True,
-                               help='AWS ARN for assumed role')
+    common_parser.add_argument('--aws-profile', default=os.environ.get('AWS_PROFILE', 'gophillygo'),
+                               help='AWS profile')
     common_parser.add_argument('--cac-config-path', default=os.path.join(file_dir, 'default.yaml'),
                                help='Path to CAC stack config')
     common_parser.add_argument('--cac-profile', default='default',
                                help='CAC stack profile to use for launching stacks')
 
-    if os.path.isfile(os.path.expanduser('~/.aws/credentials')):
```

Comment:
Not sure what bug this references, but `AWS_PROFILE` appears to be respected properly by Packer 1.6.5